cmd/launch-taskrun/konflux/ecp_lookup.go

```diff
@@ -0,0 +1,104 @@
+package konflux
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// ClientReader interface captures only the read operations we need for testability
+type ClientReader interface {
+	Get(ctx context.Context, key client.ObjectKey, obj client.Object, opts ...client.GetOption) error
+	List(ctx context.Context, list client.ObjectList, opts ...client.ListOption) error
+}
+
+// findReleasePlan looks for a release plan applicable for a given application
+func FindReleasePlan(ctx context.Context, cli ClientReader, appName string, ns string) (ReleasePlan, error) {
+	var rp ReleasePlan
+
+	// Get all release plans in the namespace
+	planList := &ReleasePlanList{}
+	err := cli.List(ctx, planList, client.InNamespace(ns))
+	if err != nil {
+		return rp, fmt.Errorf("failed to lookup release plan in namespace %s: %w", ns, err)
+	}
+	if len(planList.Items) == 0 {
+		return rp, fmt.Errorf("no release plans found in namespace %s: %w", ns, err)
+	}
+
+	// Filter to find just the release plans for the given application
+	var matchingPlans []ReleasePlan
+	for _, plan := range planList.Items {
+		if plan.Spec.Application == appName {
+			matchingPlans = append(matchingPlans, plan)
+		}
+	}
+	if len(matchingPlans) == 0 {
+		return rp, fmt.Errorf("no release plans found for application name: %s", appName)
+	}
+
+	// Choose one of the release plans
+	// TODO: I'm expecting most of the time there will be only one releasePlan, but
+	// I'm not sure how correct that is. Could there be more than one? If there was
+	// more than one, how would we know which one to choose?
```

Comment:
Could be useful to log if there were more than one found so we might have an indication if there is an error?

Reply:
Done in upcoming revision.